libcontainer/specconv/spec_linux.go

```diff
@@ -771,8 +771,14 @@ func CreateCgroupConfig(opts *CreateOpts, defaultDevs []*devices.Device) (*confi
 				c.Resources.CpusetMems = r.CPU.Mems
 				c.Resources.CPUIdle = r.CPU.Idle
 			}
+			// Convert pids limit from the runtime-spec value (where any value <= 0 means "unlimited")
+			// to internal runc value (where -1 is "unlimited", and 0 is "unset").
 			if r.Pids != nil {
-				c.Resources.PidsLimit = r.Pids.Limit
+				if r.Pids.Limit > 0 {
```

Comment:
I don't know whether upstream projects should write such codes or not when they update the container's pids limit?
```
if r.Pids.Limit > 0 {
    c.Resources.PidsLimit = r.Pids.Limit
} else {
    c.Resources.PidsLimit = -1
}
```
@AkihiroSuda Do you know what's the situation in `nerdctl`.

Reply:
Maybe in kubelet they should know this information:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cgroup_manager_linux.go#L355-L381

Reply:
So I still think we should fix it in `pids.go`, because others always use `Manager.Set` directly without use `specconv`.
But we have to solve 2 problems if we choose solutions like #4015 :
1. Don't always write `pids.max` if `PidsLimit == 0`;
2. Don't introduce break changes of libcontainer API.

Reply:
I took a look at git history, and it seems that libcontainer _always_ treated `0` as unset for pids limit, since commit db3159c9d9d3b65d2d27ff33f722eb9797f20e52 from January 2016 that added the initial pids limit support (alas, the very same commit incorrectly documents the field (https://github.com/opencontainers/runc/commit/db3159c9d9d3b65d2d27ff33f722eb9797f20e52#diff-b71b6973c045d22e41e802cf65cade82c573a844190bc0d06a6ade9f21cb2c5c), and this PR fixes that).

Thus, I am pretty sure libcontainer users are aware of what 0 means for pids limit in libcontainer; yet I am going to look into kubernetes and nerdctl to confirm.

Now, we have two problems to solve:
1. runc violates the runtime spec by treating pids.limit==0 as unset rather than unlimited (#4014);
2. libcontainer's handling of pids limit value differs from runtime spec.

I think that 2 is much less of a problem than 1, and can be solved by properly documenting the PidsLimit field (this is what this PR does).

Reply:
FWIW: case 1 is the only case I care about fixing